sc-tools/sc-machine-runner/CMakeLists.txt

```diff
@@ -1,14 +1,20 @@
 set(SC_MACHINE_RUNNER_ROOT ${CMAKE_CURRENT_LIST_DIR})
 
-file(GLOB_RECURSE SOURCES "src/*.cpp" "src/*.h" "src/*.hpp")
+file(GLOB SOURCES CONFIGURE_DEPENDS "src/*.cpp" "src/*.hpp")
 list(REMOVE_ITEM SOURCES "${SC_MACHINE_RUNNER_ROOT}/src/main.cpp")
 
 add_library(sc-machine-runner SHARED ${SOURCES})
-include_directories(${SC_MEMORY_SRC} ${SC_CONFIG_UTILS_SRC} ${GLIB2_INCLUDE_DIRS})
-target_link_libraries(sc-machine-runner sc-memory sc-config-utils)
+target_link_libraries(sc-machine-runner
+    LINK_PRIVATE sc-memory
+    LINK_PRIVATE sc-config-utils
+)
+target_include_directories(sc-machine-runner
+    PUBLIC ${SC_MEMORY_SRC}
+    PUBLIC ${SC_CONFIG_UTILS_SRC}
+)
```

Comment:
This is confusing. You link sc-memory privately, but make the include directories public. Why?